GLAA.Web/Controllers/OrganisationDetailsController.cs

```diff
@@ -95,75 +108,131 @@ public IActionResult AddPreviousTradingName(BusinessNameViewModel model)
         [ExportModelState]
-        public IActionResult SaveBusinessPhoneNumber(BusinessPhoneNumberViewModel model)
+        public IActionResult BusinessPhoneNumber(BusinessPhoneNumberViewModel model)
         {
-            return OrganisationDetailsPost(model, 4);
+            return OrganisationDetailsPost(model, nameof(BusinessPhoneNumber));
+        }
+
+        [HttpGet]
+        [ImportModelState]
+        public IActionResult BusinessMobileNumber(bool? back = false)
+        {
+            return OrganisationDetailsGet(nameof(BusinessMobileNumber), back);
         }
 
         [HttpPost]
         [ExportModelState]
-        public IActionResult SaveBusinessMobileNumber(BusinessMobileNumberViewModel model)
+        public IActionResult BusinessMobileNumber(BusinessMobileNumberViewModel model)
         {
-            return OrganisationDetailsPost(model, 5);
+            return OrganisationDetailsPost(model, nameof(BusinessMobileNumber));
+        }
+
+        [HttpGet]
+        [ImportModelState]
+        public IActionResult BusinessEmailAddress(bool? back = false)
+        {
+            return OrganisationDetailsGet(nameof(BusinessEmailAddress), back);
         }
 
         [HttpPost]
         [ExportModelState]
-        public IActionResult SaveBusinessEmailAddress(BusinessEmailAddressViewModel model)
+        public IActionResult BusinessEmailAddress(BusinessEmailAddressViewModel model)
+        {
+            return OrganisationDetailsPost(model, nameof(BusinessMobileNumber));
+        }
+
+        [HttpGet]
+        [ImportModelState]
+        public IActionResult BusinessWebsite(bool? back = false)
         {
-            return OrganisationDetailsPost(model, 6);
+            return OrganisationDetailsGet(nameof(BusinessWebsite), back);
         }
 
         [HttpPost]
         [ExportModelState]
-        public IActionResult SaveBusinessWebsite(BusinessWebsiteViewModel model)
+        public IActionResult BusinessWebsite(BusinessWebsiteViewModel model)
+        {
+            return OrganisationDetailsPost(model, nameof(BusinessWebsite));
+        }
+
+        [HttpGet]
+        [ImportModelState]
+        public IActionResult CommunicationPreference(bool? back = false)
         {
-            return OrganisationDetailsPost(model, 7);
+            return OrganisationDetailsGet(nameof(CommunicationPreference), back);
         }
 
         [HttpPost]
         [ExportModelState]
-        public IActionResult SaveCommunicationPreference(CommunicationPreferenceViewModel model)
+        public IActionResult CommunicationPreference(CommunicationPreferenceViewModel model)
         {
-            return OrganisationDetailsPost(model, 8);
+            return OrganisationDetailsPost(model, nameof(CommunicationPreference));
+        }
+
+        [HttpGet]
+        [ImportModelState]
+        public IActionResult LegalStatus(bool? back = false)
+        {
+            return OrganisationDetailsGet(nameof(LegalStatus), back);
```

Comment:
The more I see these get actions the more I think that's totally fine. I don't we can keep things totally generic and we're finding more and more edge cases and complexity with trying to centralise it all. 
As a result, fully support breaking these down to separate actions.

Reply:
Agreed, it'll give us a lot more flexibility going forward.